server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetaData.java

```diff
@@ -0,0 +1,291 @@
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastCommittedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[1];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static VotingConfiguration lastAcceptedConfig(Object[] termAndConfig) {
+        List<String> nodeIds = (List<String>) termAndConfig[2];
+        return new VotingConfiguration(new HashSet<>(nodeIds));
+    }
+
+    private static final ConstructingObjectParser<CoordinationMetaData, Void> PARSER = new ConstructingObjectParser<>(
+            "coordination",
+            termAndConfigs -> new CoordinationMetaData(term(termAndConfigs), lastCommittedConfig(termAndConfigs),
+                    lastAcceptedConfig(termAndConfigs), Collections.emptySet()));
+    static {
+        PARSER.declareLong(ConstructingObjectParser.constructorArg(), TERM_PARSE_FIELD);
+        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), LAST_COMMITTED_CONFIGURATION_FIELD);
+        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), LAST_ACCEPTED_CONFIGURATION_FIELD);
+    }
+
+    public CoordinationMetaData(long term, VotingConfiguration lastCommittedConfiguration, VotingConfiguration lastAcceptedConfiguration,
+        Set<DiscoveryNode> votingTombstones) {
+        this.term = term;
+        this.lastCommittedConfiguration = lastCommittedConfiguration;
+        this.lastAcceptedConfiguration = lastAcceptedConfiguration;
+        this.votingTombstones = Collections.unmodifiableSet(new HashSet<>(votingTombstones));
+    }
+
+    public CoordinationMetaData(StreamInput in) throws IOException {
+        term = in.readLong();
+        lastCommittedConfiguration = new VotingConfiguration(in);
+        lastAcceptedConfiguration = new VotingConfiguration(in);
+        votingTombstones = Collections.unmodifiableSet(in.readSet(DiscoveryNode::new));
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder builder(CoordinationMetaData coordinationMetaData) {
+        return new Builder(coordinationMetaData);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeLong(term);
+        lastCommittedConfiguration.writeTo(out);
+        lastAcceptedConfiguration.writeTo(out);
+        out.writeCollection(votingTombstones, (o, v) -> v.writeTo(o));
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        return builder
+            .field(TERM_PARSE_FIELD.getPreferredName(), term)
+            .field(LAST_COMMITTED_CONFIGURATION_FIELD.getPreferredName(), lastCommittedConfiguration)
+            .field(LAST_ACCEPTED_CONFIGURATION_FIELD.getPreferredName(), lastAcceptedConfiguration);
+        // TODO include voting tombstones here
```

Comment:
let's add this here as well. We will need it anyway if the persistence layer is to work correctly.

Reply:
We decided to make this change in follow-up PR, because we need a different class for voting tombstones, not `DiscoveryNode`.

Reply:
👍